File: trustdevice/src/main/java/cn/tongdun/mobrisk/core/collectors/DevicePersonalizationInfoCollector.java
package cn.tongdun.mobrisk.core.collectors;

import java.util.Locale;
import java.util.TimeZone;

/**
 * @description: DevicePersonalizationInfoCollector
 * @author: wuzuchang
 * @date: 2022/12/6
 */
public class DevicePersonalizationInfoCollector {

    public DevicePersonalizationInfoCollector() {
    }

    public String getLocaleCountry() {
        return Locale.getDefault().getCountry();
    }

    public String getDefaultLanguage() {
        return Locale.getDefault().getLanguage();
    }

    public String getTimezone() {
        return TimeZone.getDefault().getDisplayName();
    }
}

File: trustdevice/src/main/java/cn/tongdun/mobrisk/core/collectors/DevicePersonalizationInfoCollector.kt
package cn.tongdun.mobrisk.core.collectors

import java.util.*

/**
 * @description:Device Personalization Info
 * @author: wuzuchang
 * @date: 2023/5/15
 */
interface DevicePersonalizationInfoInterface {
    fun getLocaleCountry(): String?
    fun getDefaultLanguage(): String?
    fun getTimezone(): String?
}

class DevicePersonalizationInfoCollector : DevicePersonalizationInfoInterface {
    override fun getLocaleCountry(): String? {
        return Locale.getDefault().country
    }

    override fun getDefaultLanguage(): String? {
        return Locale.getDefault().language
    }

    override fun getTimezone(): String? {
        return TimeZone.getDefault().displayName
    }
}